tests/ops/filter/test_video_nsfw_filter.py

```diff
@@ -0,0 +1,172 @@
+        }
+        ]
+
+        dataset = Dataset.from_list(ds_list)
+        op = VideoNSFWFilter(hf_nsfw_model=self.hf_nsfw_model,
+                            score_threshold=0.1,
+                            frame_sampling_method='uniform',
+                            frame_num=3,)
+        self._run_filter(dataset, tgt_list, op)
+
+    def test_reduce_max(self):
+
+        ds_list = [{
+            'videos': [self.video1_path]
+        }, {
+            'videos': [self.video2_path]
+        }, {
+            'videos': [self.video3_path]
+        }]
+        tgt_list = [{
+            'videos': [self.video2_path]
+        }, {
+            'videos': [self.video3_path]
+        }
+        ]
+
+        dataset = Dataset.from_list(ds_list)
+        op = VideoNSFWFilter(hf_nsfw_model=self.hf_nsfw_model,
+                            score_threshold=0.9,
+                            frame_sampling_method='all_keyframes',
+                            reduce_mode='max')
+        self._run_filter(dataset, tgt_list, op)
+
+    def test_reduce_min(self):
+
+        ds_list = [{
+            'videos': [self.video1_path]
+        }, {
+            'videos': [self.video2_path]
+        }, {
+            'videos': [self.video3_path]
+        }]
+        tgt_list = [{
+            'videos': [self.video1_path]
+        }, {
+            'videos': [self.video3_path]
+        }
+        ]
+
+        dataset = Dataset.from_list(ds_list)
+        op = VideoNSFWFilter(hf_nsfw_model=self.hf_nsfw_model,
+                            score_threshold=0.0004,
+                            frame_sampling_method='all_keyframes',
+                            reduce_mode='min')
+        self._run_filter(dataset, tgt_list, op)
+
+    def test_any(self):
+
+        ds_list = [{
+            'videos': [self.video1_path, self.video3_path]
+        }, {
+            'videos': [self.video1_path, self.video2_path]
+        }]
+        tgt_list = [{
+            'videos': [self.video1_path, self.video3_path]
+        }]
+
+        dataset = Dataset.from_list(ds_list)
+        op = VideoNSFWFilter(hf_nsfw_model=self.hf_nsfw_model,
+                            score_threshold=0.01,
+                            frame_sampling_method='all_keyframes',
+                            any_or_all='any')
+        self._run_filter(dataset, tgt_list, op)    
+
+    def test_all(self):
+
+        ds_list = [{
+            'videos': [self.video1_path, self.video2_path]
+        }, {
+            'videos': [self.video2_path, self.video3_path]
+        }]
+        tgt_list = [{
+            'videos': [self.video2_path, self.video3_path]
+        }]
+
+        dataset = Dataset.from_list(ds_list)
+        op = VideoNSFWFilter(hf_nsfw_model=self.hf_nsfw_model,
+                            score_threshold=0.1,
+                            frame_sampling_method='all_keyframes',
+                            any_or_all='all')
+        self._run_filter(dataset, tgt_list, op)   
+
```

Comment:
Do we need to test the op in the parallel mode (np > 1)?

Reply:
test in multi process passed